index.js

```diff
@@ -72,22 +71,25 @@ function findAllDependencies(file, knownDependencies, sourceDirectories, knownFi
 }
 
 // Given a source directory (containing top-level Elm modules), locate the
-// elm-package.json file that includes it.
-function findElmPackage(baseDir, currentDir) {
+// elm-package.json file that includes it and get all its source directories.
+function getElmPackageSourceDirectories(baseDir, currentDir) {
   if (typeof currentDir === "undefined") {
     currentDir = baseDir;
   }
 
   var elmPackagePath = path.join(currentDir, 'elm-package.json');
   if (fs.existsSync(elmPackagePath)) {
-    return elmPackagePath;
+    var sourceDirectories = getSourceDirectories(elmPackagePath);
+    if (_.includes(sourceDirectories, baseDir)) {
```

Comment:
Correct, and that’s supported by my implementation already. If you have:

```
my-package
`- elm-package.json ("source-directories": ["./src"],)
`- src
    `- Main.elm (module Main exposes (..))
```

…then `getBaseDir('/path/to/src/Main.elm')` will return `/path/to/src`, and then `getElmPackageSourceDirectories('/path/to/src')` will find elm-package.json in the parent directory, confirm it includes `./src` in `"source-directories"` and return `['/path/to/src']`.

Unless I’m missing something?